File: AEDS2/Verde/TP1/Palindromo.java
public class Palindromo {
    
    // Método para comparar duas strings e verificar se são iguais
    public static boolean equalStrings(String str_1, String str_2) {
        if(str_1.length() != str_2.length()) {
            return false;
        }

        for(int i = 0; i < str_1.length(); i++) {
            if(str_1.charAt(i) != str_2.charAt(i)) {
                return false;
            }
        }
        return true;
    }
    
    // Método para verificar se uma string é um palindromo ou não
    public static boolean isPalindromo(String str) {
        for(int i = 0; i < str.length() / 2; i++) {
            if(str.charAt(i) != str.charAt(str.length() - 1 - i)) {
                return false;
            }
        }
        return true;
    }

    // Main
    public static void main(String[] args) {
        String str;
        do {
            str = MyIO.readLine();
            if(!equalStrings(str, "FIM")) {
                if(isPalindromo(str)) {
                    MyIO.println("SIM");
                }
                else {
                    MyIO.println("NAO");
                }
            }
        }while(!equalStrings(str, "FIM"));
    }

}

File: AEDS2/Verde/TP1/Palindromo.c
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#include <stdio.h>

// Função para verificar se string é um palindromo.
bool isPalindromo(char string[]) {
    bool resp = true;
    int tamanho = strlen(string); 
    for(int i = 0, j = tamanho - 1; i < tamanho/2; i++, j--) {
            if(string[i] != string[j]) {
                resp = false;
                i = tamanho; 
            }
    }
    return resp; 
}

// Main
int main() {
    char *string = (char*)malloc(250 * sizeof(char)); // Aloca memoria para entrada
    bool palindromo;
    while(strcmp(string, "FIM") != 0) {
        scanf(" %[^\n]", string);  
        palindromo = isPalindromo(string); 
        if(palindromo) {
            printf("SIM\n");
        }
        else if(strcmp(string, "FIM") != 0) {
            printf("NAO\n");
        }
    }
    return 0;
}